clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ArtifactController.java

```diff
@@ -66,4 +67,63 @@ StreamingResponseBody fetch(@RequestBody Artifact artifact) {
 
     return outputStream -> IOUtils.copy(artifactDownloader.download(artifact), outputStream);
   }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/{type}/account/{accountName}/names")
+  List<String> getNames(@PathVariable("type") String type,
+                        @PathVariable("accountName") String accountName) {
+    switch (type) {
+      case "helm":
+        return getHelmArtifactNames(accountName);
```

Comment:
I would prefer to move away from a growing switch statement, and instead delegate to something autowired by spring to loop over all artifact credentials that can handle this, and take the first/only one that works: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/ArtifactDownloader.java#L53